java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -1220,8 +1220,35 @@ private static Table gatherJSONColumns(Schema schema, TableWithMeta twm, int emp
               columns[i] = tbl.getColumn(index).incRefCount();
             }
           } else {
-            try (Scalar s = Scalar.fromNull(types[i])) {
-              columns[i] = ColumnVector.fromScalar(s, rowCount);
+            if (types[i] == DType.LIST) {
+              Schema childSchema = schema.getChild(i);
+              Schema elementSchema = childSchema.getChild(0);
+              try (Scalar s = Scalar.listFromNull(elementSchema.asHostDataType())) {
+                columns[i] = ColumnVector.fromScalar(s, rowCount);
+              }
+            } else if (types[i] == DType.STRUCT) {
+              Schema childSchema = schema.getChild(i);
+              int numChildren = 0;
+              String[] childNames = childSchema.getColumnNames();
+              if (childNames == null) {
+                try (Scalar s = Scalar.structFromNull()) {
+                  columns[i] = ColumnVector.fromScalar(s, rowCount);
+                }
+              } else {
+                int numChildChildren = childSchema.getColumnNames().length;
+                DataType[] childChildrenTypes = new DataType[numChildChildren];
```

Comment:
Nit: It might be a bit easier to read this code if "child" was "struct" in many of these local variable names for processing a struct, e.g.: structChildrenTypes or structSchemaTypes instead of childChildrenTypes, numStructChildren instead of numChildChildren.